app/src/main/java/com/duckduckgo/app/pixels/ppropromo/RmfPixelParamsAdditionPlugin.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.pixels.ppropromo
+
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+
+@ContributesMultibinding(AppScope::class)
+class RmfPixelParamsAdditionPlugin @Inject constructor(
+    private val additionalPixelParamsDataStore: AdditionalPixelParamsDataStore,
+) : PixelParamsAdditionPlugin {
+    override fun isEligible(queryParams: Map<String, String>): Boolean {
```

Comment:
Proposal: instead of us checking here if this is eligible, I would make it that we return the campaign or origin from here, and we let the pixel interceptor to decide if eligible. That way we will be able to move this plugins into their own modules.